client/blocks/importer/wordpress/upgrade-plan/test/index.tsx

```diff
@@ -0,0 +1,206 @@
+			domain: 'test-example.wordpress.com',
+			jetpack: false,
+			launch_status: 'launched',
+			locale: 'en',
+			logo: { id: 'logo', sizes: [ '100x100' ], url: 'https://test-example.wordpress.com/logo' },
+			name: 'Test Site',
+			slug: DEFAULT_SITE_SLUG,
+			title: 'Test Site',
+		},
+		isBusy: false,
+		ctaText: '',
+		navigateToVerifyEmailStep: () => {},
+		onCtaClick: () => {},
+	};
+
+	return {
+		...defaultProps,
+		...customProps,
+	};
+}
+
+const mockApi = () => nock( 'https://public-api.wordpress.com:443' );
+
+const API_RESPONSE_ELIGIBLE = {
+	eligible: true,
+};
+
+const API_RESPONSE_INELIGIBLE_UNVERIFIED_EMAIL = {
+	eligible: true,
+	error_code: 'email-unverified',
+};
+
+describe( 'UpgradePlan', () => {
+	beforeAll( () => nock.disableNetConnect() );
+
+	it( 'should render the default Continue CTA and trigger onCtaClick when no CTA is supplied', async () => {
+		const mockOnCtaClick = jest.fn();
+
+		renderUpgradePlanComponent( getUpgradePlanProps( { onCtaClick: mockOnCtaClick } ) );
+
+		await userEvent.click( screen.getByRole( 'button', { name: /Continue/ } ) );
+
+		await waitFor( () => expect( mockOnCtaClick ).toHaveBeenCalled() );
+	} );
+
+	it( 'should render a custom CTA admd triggerOnCtaClick when ctaText is supplied', async () => {
+		const mockOnCtaClick = jest.fn();
+
+		renderUpgradePlanComponent(
+			getUpgradePlanProps( { ctaText: 'My Custom CTA', onCtaClick: mockOnCtaClick } )
+		);
+
+		await userEvent.click( screen.getByRole( 'button', { name: /My Custom CTA/ } ) );
+
+		await waitFor( () => expect( mockOnCtaClick ).toHaveBeenCalled() );
+	} );
+
+	it( 'should correctly trigger a Tracks event without custom event properties', async () => {
+		nock.cleanAll();
+		mockApi()
+			.persist()
+			.get(
+				`/wpcom/v2/sites/${ DEFAULT_SITE_ID }/hosting/trial/check-eligibility/${ PLAN_MIGRATION_TRIAL_MONTHLY }`
+			)
+			.reply( 200, API_RESPONSE_ELIGIBLE );
+
+		renderUpgradePlanComponent( getUpgradePlanProps() );
+
+		await waitFor( () => {
+			expect( recordTracksEvent ).toHaveBeenCalledWith(
+				'calypso_site_migration_upgrade_plan_screen',
+				{ migration_trial_hidden: 'false' }
+			);
+		} );
+	} );
+
+	it( 'should correctly trigger a Tracks event with custom event properties', async () => {
+		nock.cleanAll();
+		mockApi()
+			.persist()
+			.get(
+				`/wpcom/v2/sites/${ DEFAULT_SITE_ID }/hosting/trial/check-eligibility/${ PLAN_MIGRATION_TRIAL_MONTHLY }`
+			)
+			.reply( 200, API_RESPONSE_ELIGIBLE );
+
+		const customEventProps = {
+			custom_number: 123,
+			custom_string: 'test',
+		};
+		renderUpgradePlanComponent( getUpgradePlanProps( { trackingEventsProps: customEventProps } ) );
+
+		await waitFor( () => {
+			expect( recordTracksEvent ).toHaveBeenCalledWith(
+				'calypso_site_migration_upgrade_plan_screen',
+				{ migration_trial_hidden: 'false', ...customEventProps }
+			);
+		} );
+	} );
+
+	it( 'should correctly trigger a Tracks event that flags the trial as hidden', async () => {
```

Comment:
The test describes the usage of the track events but we are also testing if buttons are not rendered. 
Would it be better if we split it into 2 tests OR remove the extra assertions

Reply:
You're right -- split out in ab68c334b62.